motionplan/check.go

```diff
@@ -130,88 +113,105 @@ func checkPlanRelative(
 		return poseInWorld, nil
 	}
 
-	// Check out path pose is valid
-	stepEndPiF, ok := plan.Path()[wayPointIdx][checkFrame.Name()]
-	if !ok {
-		return errors.New("check frame given not in plan Path map")
-	}
-	expectedArcEndInWorld, err := toWorld(stepEndPiF, plan.Trajectory()[wayPointIdx])
+	plan := executionState.Plan()
+	zeroPosePIF := referenceframe.NewPoseInFrame(checkFrame.Name(), spatialmath.NewZeroPose())
+
+	// determine plan's starting pose
+	planStartPoseWorld, err := toWorld(zeroPosePIF, plan.Trajectory()[0])
 	if err != nil {
 		return err
 	}
 
-	currentPoseInWorld, err := toWorld(currentPoseIF, currentInputs)
+	// determine plan's ending pose
+	planEndPoseWorld, err := toWorld(zeroPosePIF, plan.Trajectory()[len(plan.Path())-1])
 	if err != nil {
 		return err
 	}
 
-	arcInputs, ok := plan.Trajectory()[wayPointIdx][checkFrame.Name()]
-	if !ok {
-		return errors.New("given checkFrame had no inputs in trajectory map at current index")
+	// setup the planOpts. Poses should be in world frame. This allows us to know e.g. which obstacles may ephemerally collide.
+	if sfPlanner.planOpts, err = sfPlanner.plannerSetupFromMoveRequest(
+		planStartPoseWorld.Pose(),
+		planEndPoseWorld.Pose(),
+		plan.Trajectory()[0],
+		worldState,
+		nil,
+		nil, // no pb.Constraints
+		nil, // no plannOpts
+	); err != nil {
+		return err
 	}
-	fullArcPose, err := checkFrame.Transform(arcInputs)
+	// change from 60mm to 30mm so we have finer interpolation along segments
+	sfPlanner.planOpts.Resolution = 30
```

Comment:
sgtm, I will put this in `check.go` since this is the only place in our code where we use this value.
I considered putting this in the constants section of `plannerOptions.go`, but ultimately decided against it for the reason above.

Reply:
Agree